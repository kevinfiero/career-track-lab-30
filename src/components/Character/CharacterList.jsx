import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const CharacterList = ({ characters }) => (

  <>
    <Character 
      image={characters[0].image}
      name={characters[0].name}
    />
    <Character 
      image={characters[1].image}
      name={characters[1].name}
    />
  </>


);

CharacterList.propTypes = {
  characters: PropTypes.array.isRequired
};

export default CharacterList;
