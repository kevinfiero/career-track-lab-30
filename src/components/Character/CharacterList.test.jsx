import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <CharacterList 
        characters = {
          [{
            image: `https://vignette.wikia.nocookie.net/animalcrossing/images/
              e/ed/Admiral_NH.png/revision/latest?cb=20200802081138`,
            name: 'Admiral',
            id: '5f5fb4bbbfd05c2aed82e460'
          },
          {
            image: `https://vignette.wikia.nocookie.net/animalcrossing/images/
              a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526`,
            name: 'Agent S',
            id: '5f5fb4bbbfd05c2aed82e461'
          }
          ]}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
