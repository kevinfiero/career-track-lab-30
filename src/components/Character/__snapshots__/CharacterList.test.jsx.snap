// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <figure>
    <img
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/
              e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
    />
    <figcaption>
      <h1>
        Admiral
      </h1>
    </figcaption>
  </figure>
  <figure>
    <img
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/
              a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
    />
    <figcaption>
      <h1>
        Agent S
      </h1>
    </figcaption>
  </figure>
</DocumentFragment>
`;
